src/components/ChatComponent/ChatComponent.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState } from 'react';
+import ChatBodySection from './Sections/ChatBodySection';
+import ChatHeaderSection from './Sections/ChatHeaderSection';
+import ChatQueueSection from './Sections/ChatQueueSection';
+import ChatInputSection from './Sections/ChatInputSection';
+import { IGetMessage, ISocketMessage, ITextMessage } from '../../interfaces';
+import {
+  createGenerateRoomMessage,
+  createGetQueueMessage,
+} from '../../services/message-service';
+import { CHAT_URL } from '../../config';
+
+const ChatComponent = () => {
+  const [socket, setSocket] = useState(null as any);
+  const [messages, setMessages] = useState([] as ITextMessage[]);
+  const [roomID, setRoomID] = useState('' as string);
+  const [uniqueID, setUniqueID] = useState('' as string);
+  const [queue, setQueue] = useState([] as string[]);
```

Comment:
I stedet for å caste alle disse verdiene kan du ta inn typen slik:
```suggestion
  const [queue, setQueue] = useState<string[]>([]);
```